rm/rm-client/src/main/java/org/ow2/proactive/resourcemanager/common/util/RMListenerProxy.java

```diff
@@ -161,43 +173,52 @@ public void rmEvent(RMEvent event) {
     /**
      * @see org.ow2.proactive.resourcemanager.frontend.RMEventListener#nodeSourceEvent(org.ow2.proactive.resourcemanager.common.event.RMNodeSourceEvent)
      */
+    @ImmediateService
     public void nodeSourceEvent(RMNodeSourceEvent event) {
-        switch (event.getEventType()) {
-            case NODESOURCE_DEFINED:
-                rmInitialState.nodeSourceAdded(event);
-                break;
-            case NODESOURCE_CREATED:
-            case NODESOURCE_SHUTDOWN:
-                rmInitialState.nodeSourceStateChanged(event);
-                break;
-            case NODESOURCE_REMOVED:
-                rmInitialState.nodeSourceRemoved(event);
-                break;
-        }
-        checkCounter(event);
+        Lambda.lock(lockState.writeLock(), () -> {
+            switch (event.getEventType()) {
+                case NODESOURCE_DEFINED:
+                    rmInitialState.nodeSourceAdded(event);
+                    break;
+                case NODESOURCE_CREATED:
+                case NODESOURCE_SHUTDOWN:
+                    rmInitialState.nodeSourceStateChanged(event);
+                    break;
+                case NODESOURCE_REMOVED:
+                    rmInitialState.nodeSourceRemoved(event);
+                    break;
+            }
+            checkCounter(event);
+        });
     }
 
     /**
      * @see org.ow2.proactive.resourcemanager.frontend.RMEventListener#nodeEvent(org.ow2.proactive.resourcemanager.common.event.RMNodeEvent)
      */
+    @ImmediateService
     public void nodeEvent(RMNodeEvent event) {
-        switch (event.getEventType()) {
-            case NODE_REMOVED:
-                rmInitialState.nodeRemoved(event);
-                break;
-            case NODE_ADDED:
-                rmInitialState.nodeAdded(event);
-                break;
-            case NODE_STATE_CHANGED:
-                rmInitialState.nodeStateChanged(event);
-                break;
+        Lambda.lock(lockState.writeLock(), () -> {
+            switch (event.getEventType()) {
+                case NODE_REMOVED:
+                    rmInitialState.nodeRemoved(event);
+                    break;
+                case NODE_ADDED:
+                    rmInitialState.nodeAdded(event);
+                    break;
+                case NODE_STATE_CHANGED:
+                    rmInitialState.nodeStateChanged(event);
+                    break;
 
-        }
-        checkCounter(event);
+            }
+            checkCounter(event);
+        });
     }
 
+    @ImmediateService
     public RMInitialState getRMInitialState(long filter) {
-        return rmInitialState.cloneAndFilter(filter);
+        return Lambda.lock(lockState.readLock(), () -> {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove useless curly braces around statement and then remove useless return keyword [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1602)